File: ISL.Reidentification.WebApps.Configurations/ISL.Reidentification.Configurations.Client/src/models/lookups/lookup.ts
import { Guid } from 'guid-typescript';

export class Lookup {
    public id: Guid;
    public lookupType: string;
    public name: string;
    public value: string;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;

    constructor(lookup: any) {
        this.id = lookup.id ? Guid.parse(lookup.id) : Guid.parse(Guid.EMPTY);
        this.lookupType = lookup.lookupType;
        this.name = lookup.name || "";
        this.value = lookup.value || "";
        this.createdDate = lookup.createdDate ? new Date(lookup.createdDate) : undefined;
        this.createdBy = lookup.createdBy || "";
        this.updatedDate = lookup.updatedDate ? new Date(lookup.updatedDate) : undefined;
        this.updatedBy = lookup.updatedBy || "";
    }
}

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server.Tests.Acceptance/Models/Lookups/Lookup.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Configurations.Server.Tests.Acceptance.Models.Lookups
{
    public class Lookup
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Value { get; set; }
        public int SortOrder { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
    }
}
